src/panels/lovelace/cards/hui-thermostat-card.ts

```diff
@@ -0,0 +1,628 @@
+     color: var(--primary-text-color);
+   }
+   .full .rs-tooltip {
+     top: 45%;
+     left: 50%;
+   }
+   .bottom .rs-tooltip {
+     top: 0;
+   }
+   .top .rs-tooltip {
+     bottom: 0;
+   }
+   .right .rs-tooltip {
+     left: 0;
+   }
+   .left .rs-tooltip {
+     right: 0;
+   }
+   .half.top .rs-tooltip,
+   .half.bottom .rs-tooltip {
+     left: 50%;
+   }
+   .half.left .rs-tooltip,
+   .half.right .rs-tooltip {
+     top: 50%;
+   }
+   .rs-tooltip .rs-input {
+     outline: 0 none;
+     border: none;
+     background: transparent;
+   }
+   .rs-tooltip-text {
+     font-size: ${Math.round(this.clientWidth / 6)}px;
+     border-radius: 7px;
+     text-align: center;
+   }
+   .rs-tooltip.edit,
+   .rs-tooltip .rs-input {
+     padding: 5px 8px;
+   }
+   .rs-tooltip.hover,
+   .rs-tooltip.edit:hover {
+     border: 1px solid #aaaaaa;
+     cursor: pointer;
+   }
+   .rs-readonly .rs-tooltip.edit:hover {
+     border-color: transparent;
+     cursor: default;
+   }
+
+   /*** handle types ***/
+   .rs-handle {
+     border-radius: 1000px;
+     outline: 0 none;
+     float: left;
+   }
+   .rs-handle.rs-handle-square {
+     border-radius: 0px;
+   }
+   .rs-handle-dot {
+     border: 1px solid #aaaaaa;
+     padding: 6px;
+   }
+   .rs-handle-dot:after {
+     display: block;
+     content: "";
+     border: 1px solid #aaaaaa;
+     height: 100%;
+     width: 100%;
+     border-radius: 1000px;
+   }
+
+   /*** theming - colors ***/
+   .rs-seperator {
+     border: 1px solid #aaaaaa;
+   }
+   .rs-border {
+     border: 1px solid #aaaaaa;
+   }
+   .rs-path-color {
+     background-color: #ffffff;
+   }
+   .rs-range-color {
+     background-color: #54bbe0;
+   }
+   .rs-bg-color {
+     background-color: #ffffff;
+   }
+   .rs-handle {
+     background-color: #838383;
+   }
+   .rs-handle-dot {
+     background-color: #ffffff;
+   }
+   .rs-handle-dot:after {
+     background-color: #838383;
+   }
+
+    </style>
+    `;
+  }
+
+  private tooltip(e) {
+    const stateObj = this.hass!.states[this.config!.entity];
+    $("#set-temperature", this.shadowRoot).text(e.value);
+    return `<div class="title">Upstairs</div>
+      <div class="current-temperature"><span class="current-temperature-text">${
+        stateObj.attributes.current_temperature
+      }</span><span class="uom">${
+      stateObj.attributes.unit_of_measurement
+    }</span></div>`;
+  }
+
+  private setTemperature(e) {
+    this.hass!.callService("climate", "set_temperature", {
+      entity_id: this.config!.entity,
+      temperature: e.value,
+    });
+  }
+
+  private renderIcon(mode, stateObj) {
+    return html`<ha-icon
+      class="${stateObj.current === mode ? mode : ""}"
+      .mode="${mode}"
+      .entity="${stateObj.entity}"
+      @click="${this.handleModeClick}"
+      icon="${modeIcons[mode]}"
+    ></ha-icon>`;
+  }
+
+  private handleModeClick(e: MouseEvent) {
+    thermostatState.attributes.operation_mode = e.currentTarget!.id;
```

Comment:
we shouldn't do this. Source of truth is always backend. We can do a temporary 2 second override while the service is being called.

Reply:
This was for my testing purpose and fake hass

Reply:
All entity components have demo platforms in Home Assistant for UI development. Add this to your config:

```
climate:
  platform: demo
```